File: lib/5_2/flutter/widgets/sizedBox.dart
import 'package:flua/5_2/context.dart';
import 'package:flua/5_2/flutter/syntheticBox.dart';
import 'package:flua/5_2/table.dart' as l;
import 'package:flutter/material.dart';

loadSizedBox(l.Table table) {
  table["sizedBox"] = makeLuaDartFunc(func: (List<dynamic> args) {
    return [
      SizedBox(
        key: maybeUnwrapAndBuildArgument(args[0]["key"]),
        width: args[0]["width"],
        height: args[0]["height"],
        child: maybeUnwrapAndBuildArgument(args[0]["child"]),
      )
    ];
  });
}

File: runtime/flutter/widgets/sizedBox.ts
import { Type } from "../../dart/core/type";
import { RuntimeBaseClass } from "../../runtimeBaseClass";
import { Key } from "./../foundation/key";
import { Widget } from "../widget";
import { StatelessWidget } from "./statelessWidget";

export interface SizedBoxProps {
    key?: Key | undefined;
    width?: number | undefined;
    height?: number | undefined;
    child?: Widget | undefined;
}

declare const flutter: {
    widgets: {
        sizedBox: (this: void, props: SizedBoxProps) => { tag: string };
    };
};

export class SizedBox extends StatelessWidget implements RuntimeBaseClass {
    public readonly internalRuntimeType = new Type(SizedBox);
    public props: SizedBoxProps;
    public constructor(props: SizedBoxProps) {
        super();
        this.props = props;
    }

    public build(): Widget {
        return flutter.widgets.sizedBox(this.props);
    }
}
